authentication/AuthenticationExample/MainViewController.m

```diff
@@ -82,6 +83,7 @@
 
 @interface MainViewController ()
 @property(strong, nonatomic) FIRAuthStateDidChangeListenerHandle handle;
+@property(strong, nonatomic) FIROAuthProvider *microsoftProvider;
```

Comment:
Seems no need to store it as a property.

Reply:
OAuthProvider is different than other providers. Others can sign in use class (static) method. OAuth sign in is an instance method. So an object needs to be generated and we need to store a pointer somewhere. Otherwise, when running the block call, it will find the weak self pointer is already deallocated.

Reply:
SGTM.